### PR TITLE
Add KubeVirt platform type

### PR DESCRIPTION
**Pull request by nirarg**
This PR is the first step of Adding KubeVirt as a platform provider for Openshift.
This change is required for adding KubeVirt platform to machine-api-operator, relevant changes (work in progress) can be found under: https://github.com/nirarg/machine-api-operator/tree/kubevirt-provider

### CODE REVIEW

File: config/v1/0000_10_config-operator_01_infrastructure.crd.yaml [hunk 1]
```diff
@@ -141,6 +141,7 @@ spec:
                   - VSphere
                   - oVirt
                   - IBMCloud
+                  - Kubevirt
```
Comment (soltysh) [inline config/v1/0000_10_config-operator_01_infrastructure.crd.yaml:144]: KubeVirt, be consistent everywhere.

Comment (nirarg) [reply config/v1/0000_10_config-operator_01_infrastructure.crd.yaml]: Thanks @soltysh, Changed it to KubeVirt

**Comment by nirarg** [comment]
Hi, can anyone review this?

**Pull request closed without merging**